languages/csharp/concept-exercises/pattern-matching/.docs/hints.md

```diff
@@ -0,0 +1,28 @@
+﻿# Hints
+
+### 1. Check the type 
+- This exercise is based on [`pattern match` match](https://docs.microsoft.com/en-us/dotnet/csharp/pattern-matching)
+so be sure to look it over carefully.
+
+### 2. Pattern Match the types 
+
+- Remember you are matching based on what object is sent into the 
+method. This means if someone submits a Square
+object you should have a pattern that catches it
+and evaluates it.
+
+### 3. Implement perimeter math
+ 
+- In this case you just need to return the right 
+number that you would get after evaluating
+the object. The input is given for you in the test
+so all you need to do is complete the math based upon
+the objects classes.
+
+### 4. Check for edge cases
+
+- Make sure you check that the method is actually
+getting a value
+- You can do this several ways [Nullable Types](https://stackoverflow.com/questions/271588/passing-null-arguments-to-c-sharp-methods)
+- [Optional Arguments](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/named-and-optional-arguments)
```

Comment:
Is it necessary for this exercise to use optional arguments? If not, I'd prefer removing them as they make for an additional requirement. In general, we want our exercises to have the bare minimal number of prerequisites.